### DISCUSSION TITLE
Open issues if the response times are too high

### DISCUSSION THREAD

**Discussion by maximousblk** [upptime/upptime#72] [Ideas] [answered]
Similar to the current "🛑 Endpoint is down" issues, a warning issue for large response times saying ":warning: Endpoint is experiencing degraded performance" would be nice.

**Comment by AnandChowdhary** [comment]
Sounds like a good idea!

**Comment by AnandChowdhary** [comment, marked as answer]
This is now added in [upptime/uptime-monitor v1.15.0](https://github.com/upptime/uptime-monitor/releases/tag/v1.15.0). You can set the `maxResponseTime` of each site like so:

```yaml
sites:
  - name: Example
    url: https://example.com
    maxResponseTime: 5000
```

In the above example, `5000` [ms] means that if the response time is greater than 5 seconds, mark the site as "degraded performance". Some interesting changes:

- `status` (in the API, commits, README, etc.,) can now be "up", "down", or "degraded", rather than only "up" or "down"
- On the README summary of a site, you can see "🟩 Up", "🟥 Down", or "🟨 Degraded"
- On the overall summary of your README, you can see:
  - "🟩 All systems operational" if all your sites are up
  - "🟨 Degraded performance" if one or more sites are degraded
  - "🟧 Partial outage" if one or more sites are down
  - "🟥 Complete outage" if all your sites are down
- As you'd expect, an issue is opened if your site experiences degraded performance, and it's auto-closed when the response time is back under the allowed threshold
- Lastly, you can now also customize each of these messages using the configuration file

I hope this solves your use case! 😄

**Comment by maximousblk** [comment]
Wow! thanks a lot! this awesome 🚀

**Comment by maximousblk** [comment]
This doesn't seem to be documented yet

**Comment by AnandChowdhary** [comment]
Thanks for the reminder, all done!
